docs/web_quickstart.rst

```diff
@@ -196,6 +196,12 @@ First one is *optimized*. You have got the idea.
 Variable Resources
 ^^^^^^^^^^^^^^^^^^
 
+.. versionchanged:: 4.0
```

Comment:
If I read your benchmarks right, we are seeing 2+ seconds to resolve a URL on an application with 5000 resources (a lot, but not completely unreasonable). That seems completely unacceptable, so I'd be keen to release this in 3.10. I think the behaviour changes are low risk enough to introduce then.

```suggestion
.. versionchanged:: 3.10
```

Reply:
Yes, 2s on my m2 mac laptop.  Its ~6s on my production env actually 🙈

Reply:
Realistically, this is a DoS vector. I only need to send 1 request to a 404 page every 2 seconds to practically halt the server.